### DISCUSSION TITLE
force query string params order

### DISCUSSION THREAD

**Discussion by Briareos17** [livewire/livewire#2722] [3. Feature ideas]
Hi, google do not like when query string params shuffles even with the same value. It looks like duplicate page for google. It will be great to have ability to sort params in query string.

**Comment by calebporzio** [comment]
I agree.

Do you see any unintended consequences of such a change?

Like, would this break anyone's apps after they update and users still have sessions open, etc...

Also, what would the sorting strategy be? Alphabetical? Arbitrary, but kept consistent?

Do you feel able to PR this after some details are thought through?

**Reply by Briareos17** [reply]
It shouldn't brake any other apps because of simple sorting, but not changing behavior of query params beign processed.

My personal prefer to manual sort, perfectly respect params order in $queryString property.

Unfortunately I do not have such skill to make good PR for this. I even can't find where query string is building(

Also - can I just implement my own sorting logic by overloading getQueryString() or any other method, responsible for query string building?

**Comment by Briareos17** [comment]
So, does any one can help with this?

**Reply by orware** [reply]
I'm sorry I don't have Livewire internals knowledge at the moment to help out with an improvement like this but did want to share my encouragement to see it happen.

In my Discussion #2890 I noticed that when clicking on my field values in a different order, it would append the selections into the query string in the order that I was selecting them...whereas the behavior in my old web application was based on the order in the form those values were contained within, which is the same order that I have them on the page, although I would also be fine with things being displayed in the query string with the same manual order that they've been defined within the Livewire Component as an alternative to that instead...in my case I already have the query string values contained within the protected `$queryString` property in the ideal order I would want so that order would be fine to utilize...and then maybe revert to alphabetical ordering for any query string values that aren't part of that array?).
